xarray/conventions.py

```diff
@@ -543,6 +556,12 @@ def _encode_coordinates(variables, attributes, non_dim_coord_names):
                 variable_coordinates[k].add(coord_name)
                 global_coordinates.discard(coord_name)
 
+            att_val = v.attrs.get
+            if ((att_val("bounds", None) == coord_name or
+                 att_val("grid_mapping", None) == coord_name)):
```

Comment:
I'm also concerned that this is inconsistent with our other rules for CF encoding. In general, we only look at `encoding` for how to encode data on disk, not `attrs`. This suggests using `encoding={'bounds': 'something'}` as an indication that a coordinate should not be serialized as a  coordinate.

Reply:
Moved to `encoding`